File: backend/src/main/java/com/ecommerce/backend/models/Address.java
package com.ecommerce.backend.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    private String zipCode;

    @NotNull
    private String city;

    @NotNull
    private String state;

    @NotNull
    private String street;

    @NotNull
    private Integer number;

    @NotNull
    @NotBlank
    private String complement;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user", referencedColumnName = "id")
    @JsonIgnore
    @NotNull
    private User user;

}

File: frontend/ecommerce-angular/src/app/models/address.ts
export interface Address {
  id: number,
  zipCode: string,
  city: string,
  state: string,
  street: string,
  number: number,
  complement: string,
  userId: number
}

export interface CEPQuery {
  cep: string,
  uf: string,
  localidade: string,
  logradouro: string,
  complemento: string
}

export interface CityResponse {
  id: number,
  nome: string
}

export interface StateResponse {
  id: number,
  nome: string,
  sigla: string
}
